CMakeLists.txt

```diff
@@ -45,6 +45,14 @@ if(DEFINED ENV{CONDA_PREFIX})
   endif()
 endif()
 
+if(DEFINED ENV{INSTALL_PREFIX})
+  message(STATUS "Dependency install directory set to: $ENV{INSTALL_PREFIX}")
+  list(APPEND CMAKE_PREFIX_PATH "$ENV{INSTALL_PREFIX}")
+  # Allow installed package headers to be picked up before brew/system package
+  # headers
+  include_directories(BEFORE "$ENV{INSTALL_PREFIX}/include")
```

Comment:
@assignUser I agree that per target is cleaner. But INSTALL_PREFIX also has the dependencies installed via the setup script. There seems no standard way to get the INCLUDE_DIR path for each dependency. Some packages like fmt add INTERFACE_INCLUDE_DIRS to the target (fmt::fmt) property, folly on the other hand defines the `FOLLY_INCLUDE_DIRS` variable. Protobuf and CUDA are currently adding `include_directories` after resolving the dependency. Should we do that for all the dependencies with BEFORE?

Reply:
Ah, you mean for the compilation of the dependencies themselves not when we are using them. Hm, yeah `include_directories` seems to be the best way to do that due to it's global nature... though if we have `INSTALL_PREFIX` are we even building any deps from source that are not self-contained/header only?

Reply:
I meant when Velox is using the dependencies. I thought about this further and removed the global include_directories for INSTALL_PREFIX. This will only work if we always install dependencies. If some of the dependencies come from brew, this could cause issues if an older version of a package is present in INSTALL_PREFIX.

I added this to fix the fmt header issue. I now explicitly get the fmt include path and add it to include_directories with before tag. That is what you were suggesting too. Can you take another look?

Reply:
>  This will only work if we always install dependencies. 

That was my assumption `INSTALL_PREFIX` exits == all deps installed there. And in that case the solution with include_directories (or per target) would work well.

>  If some of the dependencies come from brew, this could cause issues if an older version of a package is present in INSTALL_PREFIX.

Hm,maybe but that also seems like a mis-configuration then? Either use install prefix with all required deps or don't?

I think I liked the previous version more but :shrug: no strong preferences

Reply:
> Hm,maybe but that also seems like a mis-configuration then? 

This is a good reason to use the previous approach. I will revert to that :)

Reply:
This CMake statement for header file orders breaks `duckdb` build, as `fmt` (a custom copy) and `duckdb re2` (using a different namespace) now are searching headers from `${INSTALL_PREFIX}`, instead of its local directory.

It seems also causing a build issue for `Arrow` mentioned in #11052.

We might need to rethink of a better way for the header include order, if it is really needed.